Build: Compiling fixes

AndreaRigoni opened the pull request:
Hi all, I just added some fixes because mdsplus did not compile anymore to me.

WARNING gfortran changed the policies for mismatched arguments types the code should be fixed.
I added a suppressing flag to FFLAGS as a simple workaround, see: https://gcc.gnu.org/gcc-10/porting_to.html

AndreaRigoni committed:
fix: missing typedef

AndreaRigoni committed:
fix: missing extern on variable declaration

AndreaRigoni committed:
fix: missing LIBPREFIX definition for MdsMath

AndreaRigoni committed:
fix: prevent possible empty LIBPREFIX definition

AndreaRigoni committed:
fix: make it compile in gfortran 10 ( WARNING )

WARNING gfortran changed the policies for mismatchedarguments types
the code should be fixed.
see: https://gcc.gnu.org/gcc-10/porting_to.html

AndreaRigoni committed:
fix: fortran compile flags for gcc 10 only

zack-vii commented on Makefile.inc.in:
```diff
@@ -9,7 +9,7 @@ endif
 AWK ?= @AWK@
 CC = @CC@
-CFLAGS = $(WARNFLAGS) $(NOWARNFLAGS) @CFLAGS@ -D LIBPREFIX=$(LIBPREFIX)
+CFLAGS = $(WARNFLAGS) $(NOWARNFLAGS) @CFLAGS@ -DLIBPREFIX=$(or $(LIBPREFIX),MDSplus)
```
I would have asssumed this shound break things too. the 'or' ist that a thing.

AndreaRigoni replied:
Hi Timo, the $(or "something", "else") command substitute "something" if it is non empty or "else" otherwise
should be working.. if we forgot to set LIBPREFIX it simply add MDSplus as a prefix
In any case we can remove it if you prefer

zack-vii replied:
akso i assime tge default MDSplus is only valid for one packjhage. would it not be better to add LIBPREFIX there. which os is this for.

zack-vii replied:
also, was the droppoied space intentional

AndreaRigoni replied:
it was.. but just because I always found it without spaces .. could be a compatibility thing, I will google it :-)

AndreaRigoni replied:
Timo, I thought the intent of this was to add a name for the internal version function.. however it is not save to leave the flag like -DLIBPREFIX=<empty>  so I think that MDSplus could be a solution as a generic prefix for versioning. 

Have you in mind a better solution for that? 

answering your question... the os I was compiling for is arch-linux with gcc 10.1

zack-vii commented on mdslib/Makefile.in:
```diff
@@ -14,6 +14,12 @@ MKDIR_P=@MKDIR_P@
 LIBS = @LIBS@ @GLOBUS_LIBS@ $(LIBSOCKET) @LIBRESOLV@ @LIBM@
 MDSLIB_LIBS=-lTdiShr -lTreeShr -lMdsShr
 
+# WARNING: gfortran 10 dangerous fixes ... actual fortran code must be reviewed asap
```
so this file is basically unchanged.

AndreaRigoni replied:
just a reminder because I doubt in m4 file we will remember it :-)

AndreaRigoni replied:
You are right I'll remove it

zack-vii commented on math/Makefile.in:
```diff
@@ -55,7 +57,11 @@ umach.o : umach.f
 local@LIBPRE@MdsMath = @LIBPRE@MdsMath@SHARETYPE@
 
-FFLAGS += $(FCFLAGS)
+# WARNING: gfortran 10 dangerous fixes ... actual fortran code must be reviewed asap
```
so this part is basically unchanged. can we remove the comment to eliminate the diff

AndreaRigoni replied:
just a reminder because I doubt in m4 file we will remember it :-)

AndreaRigoni replied:
You are right I'll remove it

AndreaRigoni commented:
I got some other errors in jenkins ..

AndreaRigoni commented:
I pushed the wrong button and closed the pull request .. sorry

AndreaRigoni commented:
I'm getting more and more errors in fortran.. I wonder why they were not already an issue before my changes :-/

AndreaRigoni committed:
fix:  suppress -Werror flag because of many warns

zack-vii reviewed (changes requested):
Hm well this was not the case before.. so maybe the autoconfig stuff is changing flags even for other versions. i dont think suppressing the warnings is a good procedure, especially since there were no warnings before.

AndreaRigoni commented:
I found why, it was that mdslib did not declare FFLAGS in the Makefile so it did not complain with warning because it actually skip to flag them as error at all.. now I added all the usual flags and, after them for the sole mdslib and mdsmath, I added -Wno-errors so it behaves like before.

Of course it is not good, however it permits us to compile with gcc >= 10 for now..
we should then fix the fortran crap when we will have enough passion for that. :-)

zack-vii reviewed (approved):
OK then, well we should compile this in an issue so we don't forget

AndreaRigoni commented:
Off course, I'll open it now. Thanks Timo ! :-)
